pkgs/development/tools/misc/libtree/default.nix

```
@@ -1,36 +1,42 @@
 { lib
 , stdenv
 , fetchFromGitHub
-, binutils
-, chrpath
-, cmake
-, cxxopts
-, elfio
-, termcolor
-, gtest
+, testers
+, libtree
+, runCommand
+, coreutils
+, dieHook
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "libtree";
-  version = "3.1.0";
+  version = "3.1.1";
 
   src = fetchFromGitHub {
     owner = "haampie";
     repo = "libtree";
-    rev = "v${version}";
-    sha256 = "sha256-C5QlQsBL9Als80Tv13ex2XS5Yj50Ht8eDfGYAtnh/HI=";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-q3JtQ9AxoP0ma9K96cC3gf6QmQ1FbS7T7I59qhkwbMk=";
   };
 
-  buildInputs = [ cxxopts elfio termcolor ];
-
   makeFlags = [ "PREFIX=$(out)" ];
 
-  # note: "make check" returns exit code 0 even when the tests fail.
-  # This has been reported upstream:
-  #  https://github.com/haampie/libtree/issues/77
-  nativeCheckInputs = [ gtest ];
-  checkTarget = [ "check" ];
-  doCheck = true;
+  # Fails at https://github.com/haampie/libtree/blob/v3.1.1/tests/07_origin_is_relative_to_symlink_location_not_realpath/Makefile#L28
+  doCheck = false;
+
+  passthru.tests = {
+    version = testers.testVersion {
+      package = libtree;
+      command = "libtree --version";
+      version = finalAttrs.version;
+    };
+    checkCoreUtils = self: runCommand "${finalAttrs.pname}-ls-test" {
+      nativeBuildInputs = [ self coreutils dieHook ];
+    } ''
+      ${self}/bin/libtree ${coreutils}/bin/ls > $out || die "libtree failed to show dependencies."
+      [ -s $out ]
+    '';
```

`passthru.tests` should contain derivations. Also `coreutils` in `nativeBuildInputs` are redundant and if we add the package to `PATH`, we can just use it as such:

```suggestion
    checkCoreUtils = runCommand "${finalAttrs.pname}-ls-test" {
      nativeBuildInputs = [
        finalAttrs.finalPackage
        dieHook
      ];
    } ''
      libtree ${coreutils}/bin/ls > $out || die "libtree failed to show dependencies."
      [ -s $out ]
    '';
```

---

Thanks!

---

Hmm I don't remember the context for the test thing, but jtojnar is right AFAIK.

---

Oh sorry I must have misunderstood one of your reviews...